src/promptflow/promptflow/_core/flow_execution_context.py

```diff
@@ -71,39 +73,51 @@ def invoke_tool(self, node: Node, f: Callable, kwargs):
         node_run_id = run_info.run_id
 
         traces = []
-        try:
-            hit_cache = False
-            # Get result from cache. If hit cache, no need to execute f.
-            cache_info: CacheInfo = self._cache_manager.calculate_cache_info(self._flow_id, f, [], kwargs)
-            if node.enable_cache and cache_info:
-                cache_result: CacheResult = self._cache_manager.get_cache_result(cache_info)
-                if cache_result and cache_result.hit_cache:
-                    # Assign cached_flow_run_id and cached_run_id.
-                    run_info.cached_flow_run_id = cache_result.cached_flow_run_id
-                    run_info.cached_run_id = cache_result.cached_run_id
-                    result = cache_result.result
-                    hit_cache = True
-
-            if not hit_cache:
-                Tracer.start_tracing(node_run_id, node.name)
-                result = self._invoke_tool_with_timer(node, f, kwargs)
-                traces = Tracer.end_tracing(node_run_id)
 
-            self._run_tracker.end_run(node_run_id, result=result, traces=traces)
-            # Record result in cache so that future run might reuse its result.
-            if not hit_cache and node.enable_cache:
-                self._persist_cache(cache_info, run_info)
+        with OpenTelemetryTracer.start_as_current_span(node.source.tool if node.source.tool else node.type.value):
```

Comment:
To align with the current implementation, it seems that these code should be moved to `_core/tool.py` under the definition of tool_decorator

Reply:
thanks for the hint. it's a better place.